nifi-frontend/src/main/frontend/apps/nifi/src/app/ui/common/controller-service/move-controller-service/move-controller-service.component.ts

```diff
@@ -0,0 +1,158 @@
+        MatSelectModule,
+        ControllerServiceApi,
+        ControllerServiceReferences,
+        AsyncPipe,
+        NifiSpinnerDirective,
+        NifiTooltipDirective,
+        NgTemplateOutlet
+    ],
+    styleUrls: ['./move-controller-service.component.scss']
+})
+export class MoveControllerService extends CloseOnEscapeDialog {
+    @Input() goToReferencingComponent!: (component: ControllerServiceReferencingComponent) => void;
+    protected readonly TextTip = TextTip;
+    protected controllerServiceActionProcessGroups: SelectOption[] = [];
+
+    controllerService$ = this.store.select(selectControllerService);
+
+    moveControllerServiceForm: FormGroup;
+
+    @ViewChild('stepComplete') stepComplete!: TemplateRef<any>;
+    @ViewChild('stepError') stepError!: TemplateRef<any>;
+    @ViewChild('stepInProgress') stepInProgress!: TemplateRef<any>;
+    @ViewChild('stepNotStarted') stepNotStarted!: TemplateRef<any>;
+
+    constructor(
+        @Inject(MAT_DIALOG_DATA) public request: MoveControllerServiceDialogRequest,
+        private store: Store<ControllerServiceState>,
+        private formBuilder: FormBuilder
+    ) {
+        super();
+
+        request.processGroupFlow.subscribe((flow:any) => {
+            let processGroups: SelectOption[] = [];
+            let processGroupflow = flow.processGroupFlow;
+            if (processGroupflow.breadcrumb.hasOwnProperty('parentBreadcrumb')) {
+                let parentBreadcrumb = processGroupflow.breadcrumb.parentBreadcrumb;
+                if (parentBreadcrumb.permissions.canRead && parentBreadcrumb.permissions.canWrite) {
+                    let option: SelectOption = {
+                        text: parentBreadcrumb.breadcrumb.name + ' (Parent)',
+                        value: parentBreadcrumb.breadcrumb.id
+                    }
+                    processGroups.push(option);
+                }
+            }
+
+            processGroupflow.flow.processGroups.forEach((child:any) => {
+                if (child.permissions.canRead && child.permissions.canWrite) {
+                    let option: SelectOption = {
+                        text: child.component.name,
+                        value: child.component.id
+                    }
+                    processGroups.push(option);
+                }
+            });
+
+            this.controllerServiceActionProcessGroups = processGroups;
+            if (processGroups.length > 0) {
+                this.moveControllerServiceForm.controls['processGroups'].setValue(processGroups[0].value);
+            }
+        });
+
+        // build the form
+        this.moveControllerServiceForm = this.formBuilder.group({
+            processGroups: new FormControl("Process Group", Validators.required)
+        });
+
+        this.store.dispatch(
+            setControllerService({
+                request: {
+                    controllerService: request.controllerService
+                }
+            })
+        );
```

Comment:
I don't think this component needs to store the subject Controller Service in a store.

Reply:
Removed this.